model-optimizer/mo/utils/ir_reader/layer_to_class.py

```diff
@@ -190,6 +190,13 @@ def groupconv_to_conv(op: Node):
         # We use add_destination method here to support case with multiple destinations of source port
         weights_node.in_port(0).get_source().get_connection().add_destination(op.in_port(1))
         weights_node.in_port(0).disconnect()
+    elif weights_node.type == 'Convert':
+        # Support new FP16 IRs
+        const_node = weights_node.in_port(0).get_source().node
+        assert const_node.type == 'Const', \
+            'Weights of GroupConv node {} have incorrect format'.format(op.name)
```

Comment:
I think this restriction is not needed because this case can be handled by `else` below. Just check that it is Convert node with Constant input by port 0.

Reply:
Done.